src/GitServer/pack-objects/GitCommit.ts

```diff
@@ -0,0 +1,215 @@
+// The amount of work that went into crafting these cases to handl
+// -0 (just so we don't lose that information when parsing and reconstructing)
+// but can also default to +0 was extraordinary.
+
+function simpleSign (n) {
+  return Math.sign(n) || (Object.is(n, -0) ? -1 : 1)
+}
+
+function negateExceptForZero (n) {
+  return n === 0 ? n : -n
+}
+
+function formatTimezoneOffset (minutes) {
+  let sign = simpleSign(negateExceptForZero(minutes))
+  minutes = Math.abs(minutes)
+  let hours = Math.floor(minutes / 60)
+  minutes -= hours * 60
+  let strHours = String(hours)
+  let strMinutes = String(minutes)
+  if (strHours.length < 2) strHours = '0' + strHours
+  if (strMinutes.length < 2) strMinutes = '0' + strMinutes
+  return (sign === -1 ? '-' : '+') + strHours + strMinutes
+}
+
+function parseTimezoneOffset (offset) {
+  let [, sign, hours, minutes] = offset.match(/(\+|-)(\d\d)(\d\d)/)
+  minutes = (sign === '+' ? 1 : -1) * (Number(hours) * 60 + Number(minutes))
+  return negateExceptForZero(minutes)
+}
+
+function parseAuthor (author) {
+  let [, name, email, timestamp, offset] = author.match(
+    /^(.*) <(.*)> (.*) (.*)$/
+  )
+  return {
+    name: name,
+    email: email,
+    timestamp: Number(timestamp),
+    timezoneOffset: parseTimezoneOffset(offset)
+  }
+}
+
+function normalize (str) {
+  // remove all <CR>
+  str = str.replace(/\r/g, '')
+  // no extra newlines up front
+  str = str.replace(/^\n+/, '')
+  // and a single newline at the end
+  str = str.replace(/\n+$/, '') + '\n'
+  return str
+}
+
+function indent (str) {
+  return (
+    str
+      .trim()
+      .split('\n')
+      .map(x => ' ' + x)
+      .join('\n') + '\n'
+  )
+}
+
+function outdent (str) {
+  return str
+    .split('\n')
+    .map(x => x.replace(/^ /, ''))
+    .join('\n')
+}
+
+// TODO: Make all functions have static async signature?
+
+class GitCommit {
```

Comment:
Is this something that can be a separate library? It seems like something that isomorphic-git should be doing.

Reply:
Ideally it would be something that isogit is doing, and there is a branch and [issue](https://github.com/isomorphic-git/isomorphic-git/issues/1071) for it, but it looks like progress has stalled recently.